src/main/java/io/jenkins/plugins/opentelemetry/job/jenkins/PipelineListener.java

```diff
@@ -27,6 +27,16 @@
      */
     void onStartPipeline(@Nonnull FlowNode node, @Nonnull WorkflowRun run);
 
+    /**
+     * Just before the `node` step starts.
+     */
+    void onStartNodeStep(@Nonnull StepStartNode stepStartNode, @Nonnull String nodeName, @Nonnull WorkflowRun run);
```

Comment:
I'm wondering if it is the most intuitive to call twice this event per `node(){...}` block:
* just before the build node is allocated, we say  "nodeName='Node: Allocate'" 
* just after the build node is allocated and the actual tasks can be executed, we say  "nodeName='Node: Ready'" 

I'm wondering if it could be more intuitive continue with the philosophy that each event is invoked exactly once per step, this would mean to create two distinct events in the APIs `onStartNodeStep()` and `onAfterStartNodeStep()`?
We would use the naming `onStartNodeStep()` rather than `onBeforeStartNodeStep()` to stick with our naming convention. I tried initially the more detailed convention `onBeforeStartXxx()`, `onAfterStartXxx()`,  `onBeforeEndXxx()` and `onAfterEndXxx()` but I moved away from it because it was very verbose and until this need to track the duration of node allocation, I didn't identify a need for the `onAfterStartXxx()` and `onBeforeEndXxx()` events.


```
test-pipeline
   |
   |- Phase: Start
   |
   |- Phase: Run
   |   |
   |   |- Node: Allocate, function: node, node.id: 3
   |       |
   |       |- Node: Ready, function: node, node.id: 4
   |           |
   |           |- Stage: ze-stage1, function: stage, node.id: 6
   |           |   |
   |           |   |- shell-1, function: sh, node.id: 8
   |           |
   |           |- Stage: ze-stage2, function: stage, node.id: 12
   |               |
   |               |- shell-2, function: sh, node.id: 14
   |
   |- Phase: Finalise
```